releasenotes/notes/improve-quantum-causal-cone-f63eaaa9ab658811.yaml

```diff
@@ -0,0 +1,5 @@
+---
+upgrade_circuits:
+  - |
+    Improved the method :meth:`.DAGCircuit.quantum_causal_cone` to avoid examining the same
+    non-directive node multiple times when reached from different paths.
```

Comment:
This isn't really an upgrade note, an upgrade note is reserved for something that we want to document might require user input when upgrading from the previous release to the new one. Examples would be a breaking api change, minimum dependency version bump, etc. I would expect this to be more of a feature note (to indicate improved performance) or a bug fix release note if we want to document this in the release.

Reply:
Thanks, this is not the first time I am making this mistake, but I will eventually remember what "upgrade" stands for.

Reply:
done in [23877fe](https://github.com/Qiskit/qiskit/pull/12668/commits/23877fe9fcb07837b6bba3e085597ad1c881c79e)